pkg/googlecloud/publisher.go

```diff
@@ -143,6 +174,9 @@ func (p *Publisher) Close() error {
 		return nil
 	}
 	p.closed = true
+	if p.cancelCtxClient != nil {
+		p.cancelCtxClient()
```

Comment:
Can you explain what's the reason of cancelling this in Publish?

Reply:
When the Publisher constructor is called, then the Google cloud pub sub client will open a grpc connenction behind the scenes. When the Publisher is closed, this grpc connection must be closed as well. It can be done by cancelling the context which was previously given/propagated to the grpc connection.

Reply:
Oh, makes sense, I wrongly assumed this happens in `Publish`. 🤦